File: packages/agent-sdk/src/topology.ts
/**
 * Visual topology contracts for the CaseGraph platform.
 *
 * These types describe the normalized graph structure used to render
 * agent/workflow topology in the frontend. The backend derives this
 * from the existing agent and workflow registries.
 */

// ---------------------------------------------------------------------------
// Node categories
// ---------------------------------------------------------------------------

export type TopologyNodeCategory = "agent" | "workflow" | "service";

// ---------------------------------------------------------------------------
// Nodes
// ---------------------------------------------------------------------------

export interface TopologyNode {
  id: string;
  label: string;
  category: TopologyNodeCategory;
  /** Optional short description shown in tooltips or subtitles. */
  description?: string;
  /** Category-specific metadata bag. */
  meta?: Record<string, unknown>;
}

// ---------------------------------------------------------------------------
// Edge types
// ---------------------------------------------------------------------------

export type TopologyEdgeType = "handoff" | "step" | "membership";

export interface TopologyEdge {
  id: string;
  source: string;
  target: string;
  type: TopologyEdgeType;
  label?: string;
}

// ---------------------------------------------------------------------------
// Response
// ---------------------------------------------------------------------------

export interface TopologyResponse {
  nodes: TopologyNode[];
  edges: TopologyEdge[];
}

File: packages/agent-sdk/casegraph_agent_sdk/topology.py
"""Visual topology contracts for the CaseGraph platform.

These types describe the normalized graph structure used to render
agent/workflow topology in the frontend.  The backend derives this
from the existing agent and workflow registries.
"""

from __future__ import annotations

from typing import Literal

from pydantic import BaseModel, Field


TopologyNodeCategory = Literal["agent", "workflow", "service"]
TopologyEdgeType = Literal["handoff", "step", "membership"]


class TopologyNode(BaseModel):
    id: str
    label: str
    category: TopologyNodeCategory
    description: str | None = None
    meta: dict[str, object] = Field(default_factory=dict)


class TopologyEdge(BaseModel):
    id: str
    source: str
    target: str
    type: TopologyEdgeType
    label: str | None = None


class TopologyResponse(BaseModel):
    nodes: list[TopologyNode]
    edges: list[TopologyEdge]


__all__ = [
    "TopologyEdge",
    "TopologyEdgeType",
    "TopologyNode",
    "TopologyNodeCategory",
    "TopologyResponse",
]
